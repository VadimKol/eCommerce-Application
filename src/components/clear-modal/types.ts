import { type LineItem } from '@commercetools/platform-sdk';

export interface ClearModalProps {
  isModal: boolean;
  setIsModal: (isModal: boolean) => void;
  cartItems: LineItem[];
  clearFromCart: (lineItems: string[]) => Promise<void>;
}
