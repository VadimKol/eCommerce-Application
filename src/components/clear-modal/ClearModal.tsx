import { useState } from 'react';
import { toast } from 'react-toastify';

import { CustomButton } from '../custom-button/Ð¡ustomButton';
import styles from './styles.module.scss';
import type { ClearModalProps } from './types';

export function ClearModal({ isModal, setIsModal, cartItems, clearFromCart }: ClearModalProps): JSX.Element {
  const [isClearing, setIsClearing] = useState(false);

  const handleClearing = (): void => {
    setIsClearing(true);
    clearFromCart(cartItems.map((cartItem) => cartItem.id))
      .then(() => toast('Successfully cleared', { type: 'success' }))
      .catch(() => toast('Failed to clear', { type: 'error' }))
      .finally(() => setIsModal(false));
  };

  return (
    <div
      className={isModal ? styles.wrapper : `${styles.wrapper} ${styles.wrapper_hidden}`}
      onTransitionEnd={(e) => {
        if (e.target === e.currentTarget) {
          setIsClearing(false);
        }
      }}
    >
      <div className={styles.modal}>
        <p className={styles.question}>Are you sure you want to clear the cart?</p>
        <div className={styles.buttons}>
          <CustomButton
            aria-label="Cancel"
            onClick={() => setIsModal(false)}
            isDisabled={!isModal}
            className={styles.cancel}
            variant="cancel"
          >
            Cancel
          </CustomButton>
          <CustomButton
            aria-label="Confirm"
            isDisabled={isClearing}
            onClick={handleClearing}
            className={styles.confirm}
            variant="secondary"
          >
            {isClearing ? 'Clearing' : 'Confirm'}
          </CustomButton>
        </div>
      </div>
    </div>
  );
}
