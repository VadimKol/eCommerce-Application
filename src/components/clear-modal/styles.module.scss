.wrapper {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1500;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 20px;
  background-color: rgb(0 0 0 / 40%);
  opacity: 1;
  visibility: visible;
  transition: opacity 0.3s, visibility 0.3s;
  backdrop-filter: grayscale(1);

  &_hidden {
    opacity: 0;
    visibility: hidden;
  }

  & .modal {
    width: 400px;
    padding: 30px;
    background: linear-gradient(var(--background-color) 0 0) padding-box, linear-gradient(45deg, var(--primary-color), var(--secondary-color)) border-box;
    border: 5px solid transparent;
    border-radius: 10px;

    & .question {
      margin-top: 0;
      font-size: 1.7rem;
      text-align: center;

      @media (width <= 500px) {
        font-size: 1.5rem;
      }
    }

    & .buttons {
      display: flex;
      justify-content: space-around;
      font-size: 1.5rem;

      @media (width <= 500px) {
        font-size: 1.2rem;
      }
    }

    @media (width <= 500px) {
      width: 350px;
      padding: 25px;
    }
  }
}

.confirm {
  color: var(--text-color);
  background-image: linear-gradient(45deg, #1ec04c, #17a740);

  &:disabled {
    background-image: linear-gradient(45deg, #1ec04c, #17a740);
  }
}

.cancel {
  &:disabled {
    background-image: linear-gradient(45deg, #e43737, var(--error-color));
  }
}
