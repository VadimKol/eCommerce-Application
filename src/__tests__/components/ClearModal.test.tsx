import { render } from '@testing-library/react';

import { ClearModal } from '@/components/clear-modal/ClearModal';

it('renders correctly', () => {
  const { container } = render(<ClearModal isModal setIsModal={jest.fn()} cartItems={[]} clearFromCart={jest.fn()} />);
  expect(container).toMatchSnapshot();
});
