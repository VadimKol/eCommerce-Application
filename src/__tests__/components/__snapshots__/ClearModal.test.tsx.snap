// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="modal"
    >
      <p
        class="question"
      >
        Are you sure you want to clear the cart?
      </p>
      <div
        class="buttons"
      >
        <button
          class="button button--cancel cancel"
          id=""
          type="button"
        >
          Cancel
        </button>
        <button
          class="button button--secondary confirm"
          id=""
          type="button"
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</div>
`;
